frontend/src/app/organization/organization.component.ts

```diff
@@ -30,13 +30,23 @@ export class OrganizationDetailsComponent implements OnInit {
 
   ngOnInit() {
     this.user$ = this.authService.user$;
+    this.organization$ = this.organizationService.organization$;
 
-    this.organization$ = this.activatedRoute.params
-      .switchMap(params => this.organizationService.getOrganization(params.organizationId));
+    this.activatedRoute.params
+      .switchMap(params => this.organizationService.getOrganization(params.organizationId)
+    ).subscribe();
 
     this.isUserOrgMember$ = this.organization$
       .combineLatest(this.user$, (org, user) => {
+        if (org === null || user === null) {
+          return false;
+        }
         return user.organizations.filter(organ => org.id === organ.id).length > 0;
       });
   }
+  onContact(organizationContact: OrganizationContactPayload) {
+    let organization: Organization;
+    this.organization$.take(1).subscribe(org => organization = org);
+    this.contactStatus$ = this.organizationService.sendContactForm(organization, organizationContact);
```

Comment:
shouldn't it be executed inside subscribe one line above (otherwise we would executed it before `organization` will be resolved?

@radoslawroszkowiak ?

Reply:
It does not look particularly nice, but `this.organization$.take(1).subscribe(org => organization = org);` will be blocking here. `organization` will be resolved when `this.contactStatus$ = ...` gets executed.